syntax = "proto3";

package datastorage;

service DataStorageService {
    rpc PushData (PushDataRequest) returns (PushDataResponse);
    rpc PushMeta (PushMetaRequest) returns (PushMetaResponse);
    rpc FetchMeta (FetchMetaRequest) returns (FetchMetaResponse);
    rpc FetchData (FetchDataRequest) returns (FetchDataResponse);
}

message PushDataRequest {
    string key = 1;
    bytes data = 2;
}

message PushDataResponse {
    bool success = 1;
}

message PushMetaRequest {
    string key = 1;
    bytes meta = 2;
}

message PushMetaResponse {
    bool success = 1;
}

message FetchMetaRequest {
    string key = 1;
}

message FetchMetaResponse {
    bytes meta = 1;
}

message FetchDataRequest {
    string key = 1;
}

message FetchDataResponse {
    bytes data = 1;
}